portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.js

```diff
@@ -131,7 +132,10 @@ const postUtilisationReportUpload = async (req, res) => {
       return renderPageWithError(req, res, extractDataErrorSummary, extractDataError, dueReportPeriods);
     }
 
-    const { csvValidationErrors } = await api.generateValidationErrorsForUtilisationReportData(csvJson, bankId, userToken);
+    // We filter down the JSON to only send the necessary data to the API for validation
+    const filteredCsvJson = filterReportJsonToRelevantKeys(csvJson);
+
+    const { csvValidationErrors } = await api.generateValidationErrorsForUtilisationReportData(filteredCsvJson, bankId, userToken);
```

Comment:
Can we add a unit test for this please

Reply:
Sure, what are you expecting - would you be happy with a test which spys on what we pass to the generateValidationerrors api call to check its filtered down?

Reply:
Can we perform some validation on the response before de-structuring the object, wondering in case API call fails this de-structure will fail.

Reply:
Exactly my thoughts 😄

Reply:
Added a test which hopefully covers this @ttbarnes 

The destructuring error would be caught by the try/catch and handled as a problem with service error

Reply:
Agree it will be but we wouldn't know what was the API call response in the try-catch.
Perhaps something like below:

```javascript
const response = ...
if (!response) {
throw new Error(`Empty response received from utilisation report validation ${response}`);
}
```

Above will speed up de-bugging.

Reply:
FWIW I can add this if we expect the API to not error but still return a completely empty response - I'm just trying to minimise lines of code that I don't think could ever be hit (and reduce complexity of the function). If we want to cover this case anyway then I'm happy to just chuck in the above lines of code!

Reply:
All good just wanted to make sure this case is covered but happy to leave it as it is if we can get the empty `response` in the catch trace stack.